Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -380,55 +379,47 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     TS_ASSERT(!physicalInst->isParametrized());
 
     // Check the positions of the 6 detectors in the physical instrument
-    const auto &physicalDetectorInfo = ws->detectorInfo();
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1000)),
-        V3D(0, 0, 0));
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1001)),
-        V3D(0, 1, 0));
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1002)),
-        V3D(1, 0, 0));
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1003)),
-        V3D(1, 1, 0));
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1004)),
-        V3D(2, 0, 0));
-    TS_ASSERT_EQUALS(
-        physicalDetectorInfo.position(physicalDetectorInfo.indexOf(1005)),
-        V3D(2, 1, 0));
+    TS_ASSERT_EQUALS(physicalInst->getDetector(1000)->getPos(), V3D(0, 0, 0));
```

Comment:
@raquelalvarezbanos Basically this is a workaround that the current instrument code does to support analyzers or mirrors in the secondary flight path, since there is no direct support for this. The physical instrument has detector positions as we see them, the neutronic instrument has them moved around such that map the neutrons path to a straight line from the sample.

I think you should just leave this as it is. Instrument-.20 will support non-straight beam paths (eventually), we can clean this up then.